### PR TITLE
Added functional tests for embeddable

### PR DESCRIPTION
**Pull request by iltar**
- Verified support for immutable embeddables.
- Added php 7 to build.
- Dropped php 5.4, 5.5 support.
- Making use of the [temporary database library](https://github.com/hostnet/database-test-lib).
- Added hostnet codestyle to build.
- Added Enum notation for the annotation.

### CODE REVIEW

**Comment by nicoschoenmaker** [comment]
@iltar perhaps add a section in the readme linking to the relevant samples in our functional test?

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+        - php: nightly
         - php: hhvm
     allow_failures:
+        - php: nightly
         - php: hhvm
 
-install:
-    - composer install
+before_script:
+    - travis_retry composer self-update
```
Comment (hboomsma) [inline .travis.yml]: I think, travis already does a self update.

File: composer.json [hunk 2]
```diff
@@ -4,11 +4,13 @@
     "license":           "MIT",
     "require": {
         "php":                              "5.*,>=5.6||7.*",
-        "doctrine/orm":                     "^2.4.0",
+        "doctrine/orm":                     "^2.5.4",
```
Comment (hboomsma) [inline composer.json]: Really so?

File: composer.json [hunk 3]
```diff
@@ -4,11 +4,13 @@
...
         "hostnet/entity-tracker-component": "^1.0.0||^2.0.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^5.5.0"
+        "hostnet/database-test-lib": "^1.0.2",
+        "hostnet/phpcs-tool":        "^4.0.7",
+        "phpunit/phpunit":           "^5.5"
```
Comment (hboomsma) [inline composer.json]: it actually is 5.5.__0__

File: .travis.yml [hunk 4]
```diff
@@ -1,13 +1,23 @@
...
+    - travis_retry composer --dev update
```
Comment (hboomsma) [inline .travis.yml]: see https://github.com/hostnet/entity-tracker-bundle/blob/master/.travis.yml for also testing lowest deps

**Comment by hboomsma** [comment]
@iltar few small remarks, mainly `.travis.yml` and `composer.json`.
Otherwise :+1: , if you fix those, and the build I can merge it for you.

**Pull request merged**